lib/error/lib/merr.c

```diff
@@ -95,13 +97,21 @@ merr_strerror(merr_t err, char *buf, size_t buf_sz)
     char errbuf[1024], *errmsg;
     int errno_value = merr_errno(err);
 
-    if (errno_value == EBUG)
-        return strlcpy(buf, "HSE software bug", buf_sz);
+    if (errno_value == EBUG) {
+        if (!buf)
+            return sizeof(HSE_SW_BUG_MSG) - 1;
+
+        return strlcpy(buf, HSE_SW_BUG_MSG, buf_sz);
+    }
 
     /* GNU strerror only modifies errbuf if errno_value is invalid.
      * It will only return NULL if errbuf is NULL.
      */
     errmsg = strerror_r(errno_value, errbuf, sizeof(errbuf));
+    assert(errmsg);
```

Comment:
Why can't we use caller's buffer here instead of the 1024-byte local var?

Reply:
@beckerg I think you wrote this function.

Reply:
To ensure that GNU strerror_r() will not return NULL regardless of what the caller passed in.  1024 is probably at least 4x longer than it needs to be...

Reply:
Oh is it so you can return an strlen() at the very least? Can't see any more context of the code. Makes sense to me.